app/controllers/api/v2/smart_proxies_controller.rb

```diff
@@ -59,12 +59,15 @@ def refresh
       end
 
       def version
-        begin
-          version = @smart_proxy.version
-        rescue Foreman::Exception => exception
-          render :version, :locals => {:success => false, :message => exception.message} and return
-        end
-        render :version, :locals => {:success => true, :message => version[:message]}
+        render :version, :locals => {:success => true, :result => @smart_proxy.statuses[:version].version}
+      rescue Foreman::Exception => e
+        render_error :custom_error, :status => :unprocessable_entity, :locals => {:message => e.message}
+      end
+
+      def logs
```

Comment:
This has tests actually. In `test/functional/api/v2/smart_proxies_controller_test.rb`. I also fixed version test and the interface.